### PR TITLE
refactor!: Modify command argument handling and move `on_` method to bool_command

### PR DESCRIPTION
**Pull request by Yimura**
## Changes
 - Added a `command_arguments` class to help with argument parsing
 - Moved `on_enable/on_disable` methods from `looped_command` to `bool_command`

## Fixes
 - Don't enable context menu while the cmd executor is active
 - Disable freecam and noclip movement while the other is active (as well as disable them when the cmd executor is active)

## Todo
 - [ ] ~~Re-implement arguments with `std::any`~~
 - [x] Rename pop to shift to reflect the internal functionality
 - [x] Take a large enough sample size of commands to test if they behave as expected
 - [x] Fix `bool_command::on_enable` doesn't get called on menu init

### CODE REVIEW

File: src/backend/command_arguments.hpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        template<typename T = uint64_t>
+            requires ArgumentLimit<T>
+        std::enable_if_t<std::is_pointer_v<T>, const T> pop() const
+        {
+            if (m_idx >= m_argument_count)
+            {
+                throw std::runtime_error("Attempted to pop argument beyond allocated argument size.");
+            }
+
+            return static_cast<T>(m_argument_data[m_idx++]);
```
Comment (xiaoxiao921) [inline src/backend/command_arguments.hpp]: `m_idx++`

is this correct?

File: src/backend/command_arguments.hpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        template<typename T = uint64_t>
+            requires ArgumentLimit<T>
+        T pop()
+        {
+            if (m_idx >= m_argument_count)
+            {
+                throw std::runtime_error("Attempted to pop argument beyond allocated argument size.");
+            }
+
+            return reinterpret_cast<const T&>(m_argument_data[m_idx++]);
```
Comment (xiaoxiao921) [inline src/backend/command_arguments.hpp:62]: `m_idx++`

is this correct?

File: src/backend/command_arguments.hpp (hunk 1 above)
Comment (Yimura) [reply src/backend/command_arguments.hpp]: Will be renamed to `shift`.

File: src/backend/command_arguments.hpp (hunk 2 above)
Comment (Yimura) [reply src/backend/command_arguments.hpp:62]: Will be renamed to `shift`.

File: src/backend/backend.cpp [hunk 3]
```diff
@@ -19,6 +19,8 @@ namespace big
 {
 	void backend::loop()
 	{
+		for (auto& command : g_bool_commands)
+			command->refresh();
 		for (auto& command : g_looped_commands)
 			command->refresh();
```
Comment (xiaoxiao921) [inline src/backend/backend.cpp]: double refresh?

Comment (Yimura) [reply src/backend/backend.cpp]: One for bool commands and one for looped commands.

File: src/backend/command_arguments.hpp [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+            return static_cast<T>(m_argument_data[idx]);
+        }
+
+        template<typename T = uint64_t>
+            requires ArgumentLimit<T>
+        T shift() const
+        {
+            if (m_idx >= m_argument_count)
+            {
+                throw std::runtime_error("Attempted to pop argument beyond allocated argument size.");
```
Comment (xiaoxiao921) [inline src/backend/command_arguments.hpp]: the exception says `pop` still

**Review by xiaoxiao921 — APPROVED** [review]

**Pull request merged**